tst/EnergyPlus/unit/EarthTube.unit.cc

```diff
@@ -0,0 +1,103 @@
+//     reference solely to the software portion of its product, Licensee must refer to the
+//     software as "EnergyPlus version X" software, where "X" is the version number Licensee
+//     obtained under this License and may not use a different name for the software. Except as
+//     specifically required in this Section (4), Licensee shall not use in a company name, a
+//     product name, in advertising, publicity, or other promotional activities any name, trade
+//     name, trademark, logo, or other designation of "EnergyPlus", "E+", "e+" or confusingly
+//     similar designation, without the U.S. Department of Energy's prior written consent.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR
+// IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY
+// AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+// THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+// EnergyPlus::EarthTube Unit Tests
+
+// Google Test Headers
+#include <gtest/gtest.h>
+
+// EnergyPlus Headers
+#include "Fixtures/EnergyPlusFixture.hh"
+#include <EnergyPlus/EarthTube.hh>
+#include <DataHeatBalFanSys.hh>
+#include <EnergyPlus/UtilityRoutines.hh>
+#include <DataEnvironment.hh>
+#include "Fixtures/EnergyPlusFixture.hh"
+#include <ObjexxFCL/gio.hh>
+
+using namespace EnergyPlus;
+using namespace EnergyPlus::EarthTube;
+using namespace EnergyPlus::DataHeatBalFanSys;
+using namespace ObjexxFCL;
+using namespace DataGlobals;
+using namespace EnergyPlus::DataEnvironment;
+
+namespace EnergyPlus {
+	
+
+	TEST_F( EnergyPlusFixture, EarthTube_CheckEarthTubesInZonesTest )
+	{
+
+	// AUTHOR: R. Strand, UIUC
+	// DATE WRITTEN: June 2017
+	
+	// Set subroutine arguments
+	std::string ZoneName = "ZONE 1";
+	std::string InputName = "ZoneEarthtube";
+	bool ErrorsFound = false;
+	
+	// Allocate and set earth tube parameters necessary to run the tests
+	TotEarthTube = 3;
+	EarthTubeSys.allocate( TotEarthTube );
+	EarthTubeSys( 1 ).ZonePtr = 1;
+	EarthTubeSys( 2 ).ZonePtr = 2;
+	EarthTubeSys( 3 ).ZonePtr = 3;
+	
+	// First case--no conflicts, only one earth tube per zone (ErrorsFound = false)
+	CheckEarthTubesInZones( ZoneName, InputName, ErrorsFound );
+	EXPECT_EQ( ErrorsFound, false );
+
+	// Second case--conflict with the last earth tube and first (ErrorsFound = true)
+	EarthTubeSys( 3 ).ZonePtr = 1;
+	CheckEarthTubesInZones( ZoneName, InputName, ErrorsFound );
+	EXPECT_EQ( ErrorsFound, true );
```

Comment:
Excellent.  OK, the only thing now is to add this unit test file to the CMakeLists.txt file so that the test actually gets built and run :upside_down_face: